bundles/org.openhab.binding.upnpcontrol/src/main/java/org/openhab/binding/upnpcontrol/internal/audiosink/UpnpAudioSink.java

```diff
@@ -65,27 +63,32 @@ public String getId() {
     }
 
     @Override
-    public void process(@Nullable AudioStream audioStream)
+    protected void processAsynchronously(@Nullable AudioStream audioStream)
             throws UnsupportedAudioFormatException, UnsupportedAudioStreamException {
         if (audioStream == null) {
             stopMedia();
             return;
         }
 
-        String url = null;
-        if (audioStream instanceof URLAudioStream) {
-            URLAudioStream urlAudioStream = (URLAudioStream) audioStream;
-            url = urlAudioStream.getURL();
+        if (audioStream instanceof URLAudioStream urlAudioStream) {
+            playMedia(urlAudioStream.getURL());
         } else if (!callbackUrl.isEmpty()) {
-            String relativeUrl = audioStream instanceof FixedLengthAudioStream
-                    ? audioHTTPServer.serve((FixedLengthAudioStream) audioStream, 20)
-                    : audioHTTPServer.serve(audioStream);
-            url = String.valueOf(this.callbackUrl) + relativeUrl;
+            StreamServed streamServed;
+            try {
+                streamServed = audioHTTPServer.serve(audioStream, 20, true);
+            } catch (IOException e) {
+                throw new UnsupportedAudioStreamException(handler.getUDN() + " can only handle clonable audio streams.",
```

Comment:
The httpAudioServer does support audio stream that are not cloneable.
The IOException is launched when the servlet cannot write a copy of the stream on disk (which is done only if the stream is not Clonable and the multi time boolean is true)

Reply:
So what good message do you propose ?
Is that better:
```
throw new UnsupportedAudioStreamException(handler.getUDN() + " was not able to handle the audio stream (cache on disk failed).",
```

Reply:
Yes